Create release note for scheduler (#3149)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/release-notes/09-April-2023/scheduler-optimized.md

---

> title: "dbt Cloud more intelligently processes scheduled jobs"

Reviewer: non-blocker but i think moving the verb before the adverb sounds nicer and avoids splitting the infinitive 

```suggestion
title: "dbt Cloud processes scheduled jobs more intelligently"
```

id: "scheduler-optimized"
description: "April 2023: "
sidebar_label: "Update: Scheduler optimizes job queue"
tags: [Apr-12-2023, scheduler]
---

> The dbt Cloud Scheduler now prevents queue clog by canceling queued runs in excess of 50. The job queue can build up when the duration of a job run grows over time, usually caused by more data landing in your cloud warehouse. Once the run duration becomes longer than the frequency of the job’s schedule, runs start to queue, waiting for the long running job to finish. 

Reviewer: qq - is it dbt Cloud Scheduler or dbt Cloud scheduler? suggesting lowercase for consistency and nonblocker suggestion for clarification:

```suggestion
The dbt Cloud scheduler now prevents queue clog by canceling queued runs in excess of 50. The job queue can build up when the duration of a job run grows over time, usually caused by more data landing in your cloud warehouse. When a job takes longer than its scheduled frequency, runs start to queue and wait for the long-running job to finish. 
```

> Previously, when you scheduled a job to run more frequently than it takes to complete a single run, the job queue would get backed up, and the scheduler couldn't process all the runs, so it would start canceling runs indiscriminately. You used to have to go into dbt Cloud to manually cancel all the queued runs and change the job schedule to "unclog" the scheduler queue.

Reviewer: ```suggestion
Previously, when you scheduled a job to run more frequently than it takes to complete a single run, the job queue would get backed up.  As a result, the scheduler would cancel some of the runs randomly because it couldn't process them all.  To fix this, you had to go to dbt Cloud and manually cancel the queued runs and adjust the job schedule to "unclog" the job queue.
```

> The dbt Cloud scheduler now detects when a job is in this clogged state, and manages the queue to run more optimally -- providing a helpful message when the scheduler has taken action.

Author: ```suggestion
The dbt Cloud scheduler now detects when a job is in this clogged state, and manages the queue to run more optimally. Only one run of the job can be queued at a time, and if a newer run is queued, the older one will be canceled with an error message.
```

> The dbt Cloud scheduler now detects when a job is in this clogged state, and manages the queue to run more optimally -- providing a helpful message when the schedluer has taken action.

Reviewer: fix typo

```suggestion
The dbt Cloud scheduler now detects when a job is in this clogged state, and manages the queue to run more optimally -- providing a helpful message when the scheduler has taken action.
```

Reviewer: Based on the release note, i think we should include something about how only one job can be queued at a time - so for example, the below can be added after the word 'optimally.':

Only one run of the job can be queued at a time, and if a newer run is queued, the older one will be canceled with an error message.

---
title: "The dbt Cloud scheduler now prevents queue clog by canceling unnecessary runs of over-scheduled jobs"
id: "scheduler-optimized"
description: "April 2023: "
sidebar_label: "Update: Scheduler optimizes job queue"
tags: [Apr-12-2023, scheduler]
---

The dbt Cloud Scheduler now prevents queue clog by canceling unnecessary runs of over-scheduled jobs. 

> The duration of a job run tends to grow over time, usually caused by growing amounts of data in the warehouse. If the run duration becomes longer than the frequency of the job’s schedule, the queue will grow faster than the Scheduler can process the job’s runs, leading to a runaway queue with runs that don’t need be processed.

Author: ```suggestion
The duration of a job run tends to grow over time, usually caused by growing amounts of data in the warehouse. If the run duration becomes longer than the frequency of the job’s schedule, the queue will grow faster than the scheduler can process the job’s runs, leading to a runaway queue with runs that don’t need be processed.
```

> Previously, when a job was in this over-scheduled state, the Scheduler would stop queuing runs after 50 were already in queue. This led to a bad experience in which runs indiscriminately were canceled, and a user would have to go into dbt Cloud to manually cancel all the queued runs and change the job schedule to "unclog" the scheduler queue.

Author: ```suggestion
Previously, when a job was in this over-scheduled state, the scheduler would stop queuing runs after 50 were already in queue. This led to a poor user experience where the scheduler canceled runs indiscriminately, and you'd have to log into dbt Cloud to manually cancel all the queued runs and change the job schedule to "unclog" the scheduler queue.
```

> Now, the dbt Cloud Scheduler detects when a scheduled job is set to run too frequently and appropriately cancels runs that don’t need to be processed. Specifically, scheduled jobs can only ever have one run of the job in queue, and if a more recent run gets queued, the early queued run will get canceled with a helpful error message. Users will still need to either refactor the job so it runs faster or change the job schedule to run less often if the job often gets into an over-scheduled state.

Author: ```suggestion
Now, the dbt Cloud scheduler detects when a scheduled job is set to run too frequently and appropriately cancels runs that don’t need to be processed. Specifically, scheduled jobs can only ever have one run of the job in queue, and if a more recent run gets queued, the early queued run will get canceled with a helpful error message. Users will still need to either refactor the job so it runs faster or change the job schedule to run less often if the job often gets into an over-scheduled state.
```
